core/src/cgcloud/core/box.py

```diff
@@ -319,18 +321,21 @@ def prepare( self, ec2_keypair_globs,
         if instance_type is None:
             instance_type = self.recommended_instance_type( )
 
-        virtualization_type = self.__get_virtualization_type( instance_type, virtualization_type )
+        virtualization_types = self.__get_virtualization_types( instance_type, virtualization_type )
 
         if image_ref is not None:
             image = self.__select_image( image_ref )
         else:
-            log.info( "Looking up default image for role %s, ... ", self.role( ) )
-            image = self._base_image( virtualization_type )
-            log.info( "... found %s.", image.id )
+            for type in virtualization_types:
+                log.info( "Looking up default image for role %s and type %s, ... " % (self.role( ), type) )
```

Comment:
For log messages, don't use %. Instead of 
```
log.info( "bla %s %s" % ( 1, 2 ) )
```
use
```
log.info( "bla %s %s", 1, 2 )
```

That is shorter and more efficient since the interpolation is only performed if the log level is at or below info.